### ISSUE TITLE
WidgetVar: not available after updating version from 14.0.0 to 14.0.1

### ISSUE THREAD

**Issue by Joaobds** [primefaces/primefaces#12144]
### Describe the bug

I needed to update my PF version from 14.0.0 to 14.0.1 to fix a problem about fileUpload dragAndDrop event on 14.0.0, but after the update none of my components with "widgterVar" atributte are being available in the document. Works just fine in the 14.0.0. 

Problem I was trying to solve on 14.0.0:
![Captura de tela de 2024-06-19 13-29-08](https://github.com/primefaces/primefaces/assets/122704826/b211fff8-b18e-41ba-96b4-27aa5a8f0cd6)

New problem on 14.0.1:

![Captura de tela de 2024-06-19 13-29-56](https://github.com/primefaces/primefaces/assets/122704826/d4ffb2f4-5784-4fd6-baba-c7fa50bfa808)

### PrimeFaces edition

Community

### PrimeFaces version

14.0.2

### Theme

Poseidon

### JSF implementation

Mojarra

### JSF version

2.3

### Java version

11

### Browser(s)

Chrome, Firefox

**Comment by tandraschko** [comment]
please provide a reproducer

**Comment by Joaobds** [comment]
> please provide a reproducer

I'm sorry for the lack of information. It's my first time opening an issue. 

### Reproducer

[](<ui:composition xmlns="http://www.w3.org/1999/xhtml"
                xmlns:h="http://java.sun.com/jsf/html"
                xmlns:f="http://java.sun.com/jsf/core"
                xmlns:ui="http://java.sun.com/jsf/facelets"
                xmlns:p="http://primefaces.org/ui"
                xmlns:o="http://omnifaces.org/ui"
                template="/templates/template.xhtml">

    <ui:define name="descricao-pagina"><h:outputText value="Escola"/></ui:define>
    <ui:define name="content">
        <f:metadata>
            <f:event type="javax.faces.event.PreRenderViewEvent"
                     listener="#{escolaWebBean.init}" />
        </f:metadata>
        <div class="p-grid crud-demo">
            <div class="p-col-12">
                <div class="card">
                    <h:form id="form">
                        <p:defaultCommand target="preventEnterKeyPress"/>
                        <p:commandButton id="preventEnterKeyPress" process="@none" global="false" style="display:none;"/>
                        <p:growl id="messages" showDetail="false" />
                        <div class="products-buttonbar">
                            <div>
                                <p:commandButton value="#{msg['escola.botao.label.adicionar']}" icon="pi pi-plus" action="#{escolaWebBean.novaEscola}"
                                                 style="margin-right: .5rem" />
                            </div>
                        </div>
                        <p:dataTable id="tabelaEscola"
                                     widgetVar="dtEscolas"
                                     var="escola"
                                     value="#{escolaWebBean.todasEscolas}"
                                     reflow="true"
                                     styleClass="products-table"
                                     rowKey="#{escola.idEscola}"
                                     paginator="true"
                                     rows="10"
                                     emptyMessage="#{msg['info.nenhumRegistroEncontrado']}"
                                     selectionRowMode="add"
                                     paginatorPosition="bottom">
                            <f:facet name="header">
                                <div class="products-table-header">
                                    <span style="font-weight: bold">Escolas</span>
                                    <span class="filter-container ui-input-icon-left">
                                            <i class="pi pi-search" aria-hidden="true"/>
                                            <p:inputText id="globalFilter"  placeholder="#{msg['label.buscar']}">
                                                <p:ajax event="keyup" delay="800" oncomplete="PF('dtEscolas').filter();" process="@this" global="false"/>
                                            </p:inputText>
                                        </span>
                                </div>
                            </f:facet>
                            <p:column headerText="Ações" exportable="false" >
                                <p:commandButton icon="pi pi-pencil" action="#{escolaWebBean.alterarEscola(escola)}"
                                                 styleClass="edit-button rounded-button ui-button outlined-button" process="@this" />
                                <p:commandButton class="rounded-button ui-button outlined-button" icon="pi pi-trash" process="@this" oncomplete="PF('deleteEscolaDialog').show()" >
                                    <f:setPropertyActionListener value="#{escola}" target="#{escolaWebBean.escolaSelecionada}" />
                                </p:commandButton>
                            </p:column>
                            <p:column headerText="#{msg['escola.label.nome']}" filterBy="#{escola.nome}">
                                <h:outputText value="#{escola.nome}" />
                            </p:column>
                            <p:column headerText="#{msg['escola.label.email']}" filterBy="#{escola.escolaEmail}">
                                <h:outputText value="#{escola.escolaEmail}" />
                            </p:column>
                            <p:column headerText="#{msg['escola.label.tipo']}" filterBy="#{escola.tipoEscolaDTO.nome}">
                                <h:outputText value="#{escola.tipoEscolaDTO.nome}" />
                            </p:column>
                            <p:column headerText="#{msg['escola.label.niveisEnsino']}" sortBy="#{escola.quantidadeNiveisEnsino}"
                                      filterBy="#{escola.quantidadeNiveisEnsino}">
                                <div class="ui-g-12 flex">
                                    <div class="ui-g-2 ui-md-2 ui-lg-2" style="text-align: end">
                                        <h:outputText value="#{escola.quantidadeNiveisEnsino}" />
                                    </div>
                                    <div class="ui-g-3 ui-md-3 ui-lg-3" style="text-align: start">
                                        <p:commandLink actionListener="#{escolaWebBean.carregarNiveisEnsino(escola.escolaNiveisEscolaresDTO)}"
                                                       oncomplete="PF('niveisEnsinoDialog').show()" update="formNiveisEnsino">
                                            <em class="pi pi-eye" style="font-size: large" title="Níveis de Ensino"  />
                                        </p:commandLink>
                                    </div>
                                </div>
                            </p:column>
                            <p:column headerText="#{msg['escola.label.administradores']}" sortBy="#{escola.quantidadeAdministradoresEscola}"
                                      filterBy="#{escola.quantidadeAdministradoresEscola}">
                                <div class="ui-g-12 flex">
                                    <div class="ui-g-2 ui-md-2 ui-lg-2" style="text-align: end">
                                        <h:outputText value="#{escola.quantidadeAdministradoresEscola}" />
                                    </div>
                                    <div class="ui-g-3 ui-md-3 ui-lg-3" style="text-align: start">
                                        <p:commandLink actionListener="#{escolaWebBean.carregarAdministradoresEscola(escola.idsAdministradoresEscola)}"
                                                       oncomplete="PF('administradoresEscolaDialog').show()" update="formAdministradoresEscola">
                                            <em class="pi pi-users" style="font-size: large" title="Administradores da Escola"  />
                                        </p:commandLink>
                                    </div>
                                </div>
                            </p:column>
                            <p:ajax event="filter" process="@this" global="false"/>
                        </p:dataTable>
                    </h:form>
                    <p:dialog header="Níveis de Ensino" id="niveisEnsinoDialog" width="70%" fitViewport="true"
                              dynamic="true"
                              appendTo="@(body)"
                              closeOnEscape="true"
                              position="center"
                              onShow="PF('niveisEnsinoDialog').initPosition();"
                              widgetVar="niveisEnsinoDialog" modal="true" resizable="false">
                        <h:form id="formNiveisEnsino">
                            <div class="ui-g">
                                <div class="ui-g-12">
                                    <p:dataTable id="tabelaNiveisEnsinoEscola"
                                                 widgetVar="dtTabelaNiveisEnsinoEscola"
                                                 var="nivelEscolar"
                                                 value="#{escolaWebBean.escolaNiveisEscolares}"
                                                 reflow="true"
                                                 styleClass="products-table"
                                                 paginator="true"
                                                 rows="10"
                                                 emptyMessage="#{msg['info.nenhumRegistroEncontrado']}"
                                                 selectionDisabled="true"
                                                 paginatorPosition="bottom">
                                        <p:column headerText="Nível de Ensino" style="text-align: center" width="10%" sortBy="#{nivelEscolar.nome}">
                                            <h:outputText value="#{nivelEscolar.nome}" />
                                        </p:column>
                                    </p:dataTable>
                                </div>
                            </div>
                        </h:form>
                    </p:dialog>

                    <p:dialog header="Administradores da Escola" id="administradoresEscolaDialog" width="70%" fitViewport="true"
                              dynamic="true"
                              appendTo="@(body)"
                              closeOnEscape="true"
                              position="center"
                              onShow="PF('administradoresEscolaDialog').initPosition();"
                              widgetVar="administradoresEscolaDialog" modal="true" resizable="false">
                        <h:form id="formAdministradoresEscola">
                            <div class="ui-g">
                                <div class="ui-g-12">
                                    <p:dataTable id="tabelaAdministradoresEscola"
                                                 widgetVar="dtTabelaAdministradoresEscola"
                                                 var="administradorEscola"
                                                 value="#{escolaWebBean.administradoresEscola}"
                                                 reflow="true"
                                                 styleClass="products-table"
                                                 paginator="true"
                                                 rows="10"
                                                 emptyMessage="#{msg['info.nenhumRegistroEncontrado']}"
                                                 selectionDisabled="true"
                                                 paginatorPosition="bottom">
                                        <p:column headerText="#{msg['usuario.label.codigo.usuario']}" sortBy="#{administradorEscola.idUsuario}">
                                            <h:outputText value="#{administradorEscola.idUsuario}" />
                                        </p:column>
                                        <p:column headerText="#{msg['usuario.label.nome.usuario']}" sortBy="#{administradorEscola.nome}">
                                            <h:outputText value="#{administradorEscola.nome}" />
                                        </p:column>
                                        <p:column headerText="#{msg['usuario.label.email']}" sortBy="#{administradorEscola.email}">
                                            <h:outputText value="#{administradorEscola.email}" />
                                        </p:column>
                                        <p:column headerText="#{msg['usuario.label.data.criacao']}" sortBy="#{administradorEscola.dadosRegistroDTO.criacao}">
                                            <h:outputText value="#{escolaWebBean.convertDate(administradorEscola.dadosRegistroDTO.criacao)}" />
                                        </p:column>
                                    </p:dataTable>
                                </div>
                            </div>
                        </h:form>
                    </p:dialog>
                    <p:confirmDialog widgetVar="deleteEscolaDialog" showEffect="fade" width="300"
                                     message="Deseja realmente remover a escola?"
                                     header="Excluir Escola"
                                     severity="warn">
                        <h:form id="formRemoverEscola">
                            <p:commandButton value="#{msg['botao.sim']}" icon="pi pi-check" actionListener="#{escolaWebBean.removerEscola}" process="@this" oncomplete="PF('deleteEscolaDialog').hide()"/>
                            <p:commandButton value="#{msg['botao.nao']}" type="button" styleClass="secondary-button" icon="pi pi-times" onclick="PF('deleteEscolaDialog').hide()" />
                        </h:form>
                    </p:confirmDialog>
                </div>
            </div>
        </div>
    </ui:define>
</ui:composition>)

### Current JS Behavior:
![Captura de tela de 2024-06-19 13-58-45](https://github.com/primefaces/primefaces/assets/122704826/aed95c2d-5cc2-4693-b30e-16c22a11994a)

It appears to be something related to primefaces core.js

**Comment by melloware** [comment]
@Joaobds see this issue: https://github.com/primefaces/primefaces/issues/11314

you are missing a Locale Key probably because you are using your own custom locale that is missing keys instead of using our built in locales.

**Comment by Joaobds** [comment]
> @Joaobds see this issue: #11314
> 
> you are missing a Locale Key probably because you are using your own custom locale that is missing keys instead of using our built in locales.

Thank you guys! I changed the locale key and now everything is working fine!

**Comment by tandraschko** [comment]
@melloware we should add some real error or logging instead of this exception, when a msgkey in general isnt found

**Comment by melloware** [comment]
let me look i thought i defensively handled this but clearly not.
